### PR TITLE
Implement page access checks 

### LINKED ISSUE #65: Restrict page access to authorized users
Only users who have been logged in can view the main pages of the application. Redirect any unauthorized user to either the login page or another page that displays an unauthorized error. 

Deal with expired auth tokens on api calls. (Add a way to go to the login page and back to the page they were working on).

### PR DESCRIPTION
**Pull request by hokansom**
Redirects users who don't have an access token to the login page. (It looks like a lot of code changes because I branched off of the home-page branch since that hasn't been merged yet). Just look at the last commit

Resolves #65

### CODE REVIEW

File: client/src/components/Main/InvitesSection/Invitation/index.js [hunk 1]
```diff
@@ -63,19 +69,19 @@ const getCSSFlapTransformation = (isRotated) => {
   const rotation = `rotateX(${isRotated ? -40 : 0}deg)`;
   const trfm = `perspective(400px) ${rotation}`;
   return trfm;
-}
+};
 
-const Invitation = ({ children, isUnopened, onClick }) => {
+const Invitation = ({ invite, invite: { isUnopened }, onClick, children }) => {
```
Comment (josiahsrc) [inline client/src/components/Main/InvitesSection/Invitation/index.js]: duplicate parameter `invite`

**Review by josiahsrc — COMMENTED** [review]
This looks good to me; I like the refactor changes. I may have missed some of the page access checks, but from what I can tell it looks great. I left one comment on the invite component with a duplicate param. As soon as there's some clarification on that, I'll go ahead and approve this.

File: client/src/components/Main/InvitesSection/Invitation/index.js (hunk 1 above)
Comment (hokansom) [reply client/src/components/Main/InvitesSection/Invitation/index.js]: @josiahsrc  So the single invite is passing the entire invite prop, which is used in the onclick method. The invite : {} is destructuring invite in order to access properties of invite. Doing it this way simplifies things later so we don't have to write invite.isUnopened.

Comment (josiahsrc) [reply client/src/components/Main/InvitesSection/Invitation/index.js]: @hokansom  Oh awesome! I had no idea this syntax existed, really cool. Ok, looks good to me 😄

**Review by josiahsrc — APPROVED** [review]
lgtm

**Pull request merged**